process/process_windows.go

```diff
@@ -59,6 +59,15 @@ func NewCommand(commandLine []string, workingDirectory string, env []string, acc
 			return nil, err
 		}
 		env = *environment
+	} else {
+		// if env is nil, then cmd.Env will be nil which meansa to copy the
+		// parent process's environment, which is exactly what we want.  But if
+		// env is not nil, then the intent was to extend the current list of
+		// environment variables with those in `env`, and without an
+		// accessToken that's not possible.
```

Comment:
This is close, but not quite right. See [the process handling source code](https://github.com/golang/go/blob/go1.11.4/src/os/exec_posix.go#L39-L41) from the go standard library (go 1.11.4). It does as you say; that is, if `env` is nil, env is set to `os.Environ()`. Instead we can write this whole block like this:

```go
    var err error
    var environment *[]string
    if accessToken == 0 { 
        // Running as current user; merge env with current environment.
        environment, err = win32.MergeEnvLists(&os.Environ(), &env)
    } else {
        // Running as task user; create environment block and merge env with it.
        environment, err = win32.CreateEnvironment(&env, accessToken)
    }   
    if err != nil {
        return nil, err 
    }   
    env = *environment
```

Reply:
Done, thanks for the sample code.